From 4a5577fab7490573ceaafa6e4417cf03bca178f1 Mon Sep 17 00:00:00 2001
Message-Id: <4a5577fab7490573ceaafa6e4417cf03bca178f1.1407786415.git.agx@sigxcpu.org>
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 11 Aug 2014 21:41:51 +0200
Subject: [PATCH] Don't enforce that dev already exists

Commit 9a02651a87d0fd56e2e7eedd63921a050a42b3ec tried to make sure
the device can be looked up but that e.g. isn't true for

  ip link add vnet10 type veth peer name vnet11

where vnet10 isn't yet existent.
---
 ip/iplink.c | 2 --
 1 file changed, 2 deletions(-)

--- a/ip/iplink.c
+++ b/ip/iplink.c
@@ -571,8 +571,6 @@ int iplink_parse(int argc, char **argv,
 				duparg2("dev", *argv);
 			*dev = *argv;
 			dev_index = ll_name_to_index(*dev);
-			if (dev_index == 0)
-				invarg("Unknown device", *argv);
 		}
 		argc--; argv++;
 	}
